cvc5_pythonic_api/cvc5_pythonic.py

```diff
@@ -6751,6 +6780,43 @@ def evaluate(t):
     return m[t]
 
 
+class ProofRef:
+    """A proof tree where every proof reference corresponds to the
+    root step of a proof.  The branches of the root step are the
+    premises of the step."""
+
+    def __init__(self, solver, proof):
+        self.proof = proof
+        self.solver = solver
+
+    def __del__(self):
+        if self.solver is not None:
+            self.solver = None
+
+    def __repr__(self):
+        return obj_to_string(self)
+
+    def getRule(self):
+        """Returns the proof rule used by the root step of the proof."""
```

Comment:
Add a test.

It's great to have a standalone test, but the doctests also are important, since they become (tested) documentation.